### PR TITLE
Simplify add_***() functions (follow primitiv core library)

### PR DESCRIPTION
**Pull request by vbkaisetsu**

### CODE REVIEW

File: primitiv/_optimizer.pyx [hunk 1]
```diff
@@ -163,24 +163,29 @@ cdef class Optimizer:
...
+        order of arguments.
 
-    def add_model(self, Model model):
-        """Registers all trainable parameters in a model.
+        E.g., below lines should behave similarly (except the case of exceptions):
 
-        :param model: Model to be optimized.
-        :type model: primitiv.Model
+            >>> opt.add(a, b, c, d)
+            >>> opt.add(a, b); add(c, d)
```
Comment (odashi) [inline primitiv/_optimizer.pyx]: `add(c,d)` -> `opt.add(c,d)`

**Comment by vbkaisetsu** [comment]
Fixed.

File: primitiv/_model.pyx [hunk 2]
```diff
@@ -55,91 +55,63 @@ cdef class Model:
...
-            ...         self.add_parameter("subparam1", self.subparam1)
-            ...         self.add_parameter("subparam2", self.subparam2)
-
-        """
-        for k, v in self.__dict__.items():
-            if isinstance(v, Parameter) and v not in self.added_parameters:
-                self.add_parameter(k, v)
+        if isinstance(arg, Parameter):
+            self.wrapped.add(pystr_to_cppstr(name), (<Parameter> arg).wrapped[0])
+            self.added_parameters.append(arg)
```
Comment (odashi) [inline primitiv/_model.pyx]: Maybe `added_parameters` and `added_models` can be combined into `added`.

Comment (vbkaisetsu) [reply primitiv/_model.pyx]: Thanks. Fixed.

**Pull request merged**